### PR TITLE
chore: Expose version and latest flag in client app

### PR DESCRIPTION
**Pull request by alexandreaco**
This PR updates the app's sidebar to:
- show the current version number of the app
- show the name active network
- remove the network name badge from the app logo
- show if there is an update available
- allow the user to click on the 'update available' indicator to update the app

The app should check for an update on startup, and saves that value to the store for it to be accessible to the network display in the sidebar.

In order to test the flow, I did two things:
- create a `dev-app-update.yml` in the project root (see `dev-app-update.yml.sample`)
- fiddle with the version number in `package.json` to make the app think there's an update available! :)

Example when no update is available:
Screen Shot 2021-11-04 at 7 14 01 PM

Example when an update *is* available:
Screen Shot 2021-11-04 at 7 14 48 PM

Example of the dialog when the update 'completes' (It doesn't automatically update in a dev environment)
Screen Shot 2021-11-04 at 7 14 58 PM

### CODE REVIEW

**Comment by saminakh** [comment]
@alexandreaco I can't actually tag you in a review because you were the author to this PR 😂 If you would be willing to leave a review (i dont think you can approve it though!) that would be lovely. If @caleboller or @staylorwr doesn't have the opportunity to review and can't give it an approval I can close and re-open the PR in my name so at least you have the ability to approve (assuming it's ready for approval).

File: src/components/NetworkDisplay.vue [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<template>
```
Comment (alexandreaco) [inline src/components/NetworkDisplay.vue:1]: This component is only even called from the sidebar. Let's prepend the name with `Sidebar` to keep the components folder organized

File: src/components/NetworkDisplay.vue [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  <div class="bg-gradient-to-r from-rBlue to-rBlueDark w-60 mt-3">
+    <div class="flex items-center px-5 pb-1 pt-3 text-white text-sm">
+      <p> <span class="opacity-80">
+      Version: </span> {{ versionNumber }} </p>
```
Comment (alexandreaco) [inline src/components/NetworkDisplay.vue:5]: These small bits of text, "Version:" and "Network:", should be referenced through the `i18n` library

File: src/updater.ts [hunk 3]
```diff
@@ -1,36 +1,26 @@
-import { dialog, MessageBoxReturnValue, MenuItem } from 'electron'
+import path from 'path'
+import { dialog, MessageBoxReturnValue } from 'electron'
 import { autoUpdater } from 'electron-updater'
+import { setUpdateIsAvailable } from '@/actions/electron/general'
 
-let updater: MenuItem | null
 autoUpdater.autoDownload = false
+if (process.env.WEBPACK_DEV_SERVER_URL) {
```
Comment (alexandreaco) [inline src/updater.ts:7]: Do we need to update .envrc?

File: src/updater.ts [hunk 4]
```diff
@@ -1,36 +1,26 @@
...
-    buttons: ['Sure', 'No']
-  }).then((buttonIndex: MessageBoxReturnValue) => {
-    if (buttonIndex.response === 0) {
-      autoUpdater.downloadUpdate()
-    } else {
-      if (updater) updater.enabled = true
-      updater = null
-    }
-  })
+  setUpdateIsAvailable(true)
```
Comment (alexandreaco) [inline src/updater.ts:19]: 🎉

File: src/updater.ts [hunk 5]
```diff
@@ -1,36 +1,26 @@
...
 })
 
 autoUpdater.on('update-not-available', () => {
-  dialog.showMessageBox({
-    title: 'No Updates',
-    message: 'Current version is up-to-date.'
-  })
-  if (updater) updater.enabled = true
-  updater = null
+  setUpdateIsAvailable(false)
```
Comment (alexandreaco) [inline src/updater.ts:23]: 👯

File: src/updater.ts [hunk 6]
```diff
@@ -42,10 +32,10 @@ autoUpdater.on('update-downloaded', () => {
   })
 })
 
-// export this to MenuItem click callback
-function checkForUpdates (menuItem: MenuItem): void {
-  updater = menuItem
-  updater.enabled = false
+export const downloadUpdate = (): void => {
+  autoUpdater.downloadUpdate()
```
Comment (alexandreaco) [inline src/updater.ts:36]: So much cleaner!!

**Review by alexandreaco — COMMENTED** [review]
Thanks @saminakh! The logic looks great, I just left a lot of vue-style nits 👩‍🏫

File: src/components/NetworkDisplay.vue [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import { useRouter } from 'vue-router'
+import { useWallet } from '@/composables'
+import { getVersionNumber, downloadLatestVersion, getIsUpdateAvailable } from '@/actions/vue/general'
+
+export default defineComponent({
+  setup () {
+    const router = useRouter()
+    const { activeNetwork } = useWallet(router)
+
+    const lowercaseNetwork: ComputedRef<string | undefined> = computed(() =>
```
[the remark this answers is not in the document]
Comment (saminakh) [reply src/components/NetworkDisplay.vue:30]: Maybe it's time i stop doing radix work in vim....I wonder if VSCode has some baked-in functionality that would do some of this formatting for me 😅

File: src/updater.ts (hunk 3 above)
Comment (saminakh) [reply src/updater.ts:7]: This should be good! It's used a couple of times in `background.ts`.

**Review by staylorwr — CHANGES_REQUESTED** [review]
I'm reading through these updates now but following your instructions for dev updates I receive the following crash on booting the app with `yarn electron:serve`:
Screen Shot 2021-11-08 at 2 30 15 PM

File: README.md [hunk 8]
```diff
@@ -28,3 +28,8 @@ yarn run lint
 
 ### Customize configuration
```
Comment (staylorwr) [inline README.md:29]: Can you include instructions for setting up the dev environment files to the top of the instructions like so:

```
yarn
cp .envrc{.sample,}
cp dev-app-update.yml{.sample,}
```
under the project setup?

File: src/views/Wallet/WalletSidebarDefault.vue [hunk 9]
```diff
@@ -1,16 +1,19 @@
...
-  <div class="w-60 px-5 pt-8 pb-3 flex flex-col h-full">
-    <AppLogo class="mb-10" />
+  <div class="w-60  pt-8 flex flex-col h-full">
+    <div class="mb-10 px-5">
+      <img alt="Radix DLT Logo" src="../../assets/logo.svg" class="w-30 mr-3">
+    </div>
 
-    <div class="flex flex-col text-white hover:text-rGreen transition-colors cursor-pointer mb-4" @click="setState(true)">
+    <div class="flex flex-col text-white hover:text-rGreen transition-colors
+    cursor-pointer mb-2 px-5" @click="setState(true)">
```
Comment (staylorwr) [inline src/views/Wallet/WalletSidebarDefault.vue:8]: 🤔 can you keep this to a single line or break out the lines by attr? ie:

```
<div 
  class="
  @click="
>

File: src/components/SidebarNetworkDisplay.vue [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  setup () {
+    const router = useRouter()
+    const { activeNetwork } = useWallet(router)
+
+    const lowercaseNetwork: ComputedRef<string | undefined> = computed(() =>
+      activeNetwork.value?.toUpperCase()
+    )
+
+    const versionNumber: Ref<string> = ref('')
+    getVersionNumber().then((res) => { versionNumber.value = res })
```
Comment (staylorwr) [inline src/components/SidebarNetworkDisplay.vue:42]: This is sort a business-logic like request being fired off through a `setup` function that could rerender.  What do you think about this logic (and these values) living in the `useWallet` composable?

File: src/menu.ts [hunk 11]
```diff
@@ -1,49 +0,0 @@
-import { app, Menu, MenuItem } from 'electron'
```
Comment (staylorwr) [inline src/menu.ts:1]: 👋🏼 👍🏼

**Review by staylorwr — CHANGES_REQUESTED** [review]
A real review now :) I'm unable to test the update functionality because of the crash.  And I'd love to swap the generic electron icon with the radix logo.

**Comment by saminakh** [comment]
@staylorwr did you update `dev-app-update.yml` to have a legitimate token? I reproduced this error by leaving in `my-github-token` as the value.

Thank you for the thorough feedback! I will clean this up 🙇

File: src/components/SidebarNetworkDisplay.vue (hunk 10 above)
Comment (saminakh) [reply src/components/SidebarNetworkDisplay.vue]: I think that makes sense! Fixed 👍

**Review by staylorwr — APPROVED** [review]
💯 I was able to build a production build and test the upgrade process locally.  Looks great Samina!

**Pull request merged**